chromatic_fitting.py

```diff
@@ -78,127 +86,277 @@ def summarize_parameters(self):
+            self.sample()
+
+        try:
+            with self.pymc3_model:
+                return sample_posterior_predictive(self.trace, ndraws)
+        except:
+            posteriors = []
+            for mod, trace in zip(self.pymc3_model, self.trace):
+                with mod:
+                    posteriors.append(sample_posterior_predictive(trace, ndraws))
+            return posteriors
+
+
+    def optimize(self, **kw):
+        """
+        Wrapper for PyMC3_ext sample
+        """
+        with self.pymc3_model:
+            return optimize(**kw)
 
     def sample(self, **kw):
         """
         Wrapper for PyMC3_ext sample
         """
-        with self.model:
-            self.trace = sample(**kw)
+        if self.optimization == "separate":
+            self.trace = []
+            for mod in self.pymc3_model:
+                with mod:
+                    self.trace.append(sample(**kw))
+        else:
+            with self.pymc3_model:
+                self.trace = sample(**kw)
 
     def summarize(self, **kw):
         """
         Wrapper for arviz summary
         """
-        with self.model:
-            self.summary = summary(self.trace, **kw)
+        if not hasattr(self, 'trace'):
+            print("Sampling has not been run yet! Running now with defaults...")
+            self.sample()
+
+        if self.optimization == "separate":
+            self.summary = []
+            for mod, trace in zip(self.pymc3_model, self.trace):
+                with mod:
+                    self.summary.append(summary(trace, **kw))
+        else:
+            with self.pymc3_model:
+                self.summary = summary(self.trace, **kw)
 
     def plot_priors(self, n=3):
         """
         Plot n prior samples from the parameter distributions defined by the user
         :parameter n
         Number of priors to plot (default=3)
         """
-        data = self.get_data()
-        prior_predictive_trace = self.sample_prior(ndraws=n)
-        for i in range(n):
-            flux_for_this_sample = np.array([prior_predictive_trace[f'wavelength_{w}_data'][i] for w in range(data.nwave)])
-            data.fluxlike[f'prior-predictive-{i}'] = flux_for_this_sample
-        data.imshow_quantities()
+        # setup the models, data and orbits in a format for looping
+        if self.optimization == "separate":
+            datas = [self.get_data(i) for i in range(self.data.nwave)]
+            prior_predictive_traces = self.sample_prior(ndraws=n)
+        else:
+            datas = [self.get_data()]
+            prior_predictive_traces = [self.sample_prior(ndraws=n)]
+
+        for nm, (data, prior_predictive_trace) in enumerate(zip(datas, prior_predictive_traces)):
+            for i in range(n):
+                flux_for_this_sample = np.array(
+                    [prior_predictive_trace[f'wavelength_{w+nm}_data'][i] for w in range(data.nwave)])
+                data.fluxlike[f'prior-predictive-{i}'] = flux_for_this_sample
```

Comment:
In both `plot_priors` and `plot_posteriors`, I wonder if it might be easier to view if we just created a new copy of the Rainbow and populated its `fluxlike` arrays from scratch, to only include `flux`, `posterior/prior-predictive-1`, `posterior/prior-predictive-2`, etc...? That way everything would be directly comparable to each other. If we want to see the other fluxlike quantities that exist too, we could do that just with two separate commands? 
```
data.imshow_quantities()
model.plot_posteriors()
```
(Obviously a super minor visual nit-picky thing!)